hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -663,6 +663,12 @@ def generate_new_examples(self):
                 HealthCheck.large_base_example
             )
 
+        for i in hrange(self.cap):
+            if i not in zero_data.forced_indices:
+                break
+        else:
+            self.exit_with(ExitReason.finished)
```

Comment:
Why does this fix the bug? (What was the bug?)

Reply:
When the test case started with a write of length `self.cap <= n < buffer\_size - 1` you would end up in a situation where the attempt to generate a novel prefix for the test case would fail (triggering https://travis-ci.org/HypothesisWorks/hypothesis/jobs/399408345#L1150) because the area after `self.cap` is forced to be zero.

When in this situation you end up with only one valid test case, so the correct thing to do is to return here, which means the problematic code (which works correctly as long as there is a free index `< self.cap` can never be triggered).

Reply:
It would be nice to have a comment to this effect next to the code block for anyone trying to understand what it does (the confusion you avoid may be your own).